### PR TITLE
CalcArrayType needs to be handled like ArrayType at some places…

### PR DESCRIPTION
**Pull request by ams-tschoening**
…or things might break e.g. in case of value instances returning lists of user types.

This is related to https://github.com/kaitai-io/kaitai_struct/issues/561.

### CODE REVIEW

File: formats/repeat_until_calc_array_type.ksy [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      - id: marker
+        type: u1
+      - id: body
+        type: u4
+instances:
+  recs_accessor:
+    value: records
+
+  first_rec:
+    value: recs_accessor.first
```
Comment (GreyCat) [inline formats/repeat_until_calc_array_type.ksy]: Missing newline at EOF

**Comment by GreyCat** [comment]
Thanks for this idea too, my apologies again for missing it in the past. Merging in.

**Comment by GreyCat** [comment]
Can I ask you to go one step further and convert this spec to .kst, so we can propagate this to all other languages?

**Comment by ams-tschoening** [comment]
I gave it a try and pushed to the same branch. After running `./spec_kst_to_all` I got the following output, which looks somewhat promising?

    doAllSpecs(repeat_n_strz_double)
    doAllSpecs(repeat_until_calc_array_type)
    doAllSpecs(repeat_until_complex)

Is this KST-thing documented somewhere? Couldn't find anything and thought that automatic generation of tests was still based on Ruby-specs. Should that spec be deleted with having a KST-file in place or do you don't care? Looking at GIT-logs, I have the feeling you simply added KST files and kept what was already available.

Additionally, would be great to have some docs about how this whole setup is used to work, because I ran into the following issue at first:

	[info] Loading project definition from C:\Users\tschoening\Documents\Eclipse\Java Sm-Mtg\Libs Java 3rd\Parsers\Generators\kaitai_struct\tests\libs_java_3rd_usage\src\translator\project
	[info] Updating ProjectRef(uri("file:/C:/Users/tschoening/Documents/Eclipse/Java%20Sm-Mtg/Libs%20Java%203rd/Parsers/Generators/kaitai_struct/tests/libs_java_3rd_usage/src/translator/project/"), "translator-build")...
	[info] Done updating.
	[info] Loading settings from build.sbt ...
	[info] Set current project to translator (in build file:/C:/Users/tschoening/Documents/Eclipse/Java%20Sm-Mtg/Libs%20Java%203rd/Parsers/Generators/kaitai_struct/tests/libs_java_3rd_usage/src/translator/)
	[info] Updating ...
	[warn]  module not found: io.kaitai#kaitai-struct-compiler_2.12;0.9-SNAPSHOT
	[warn] ==== local: tried
	[warn]   C:\Users\tschoening\.ivy2\local\io.kaitai\kaitai-struct-compiler_2.12\0.9-SNAPSHOT\ivys\ivy.xml
	[warn] ==== public: tried
	[warn]   https://repo1.maven.org/maven2/io/kaitai/kaitai-struct-compiler_2.12/0.9-SNAPSHOT/kaitai-struct-compiler_2.12-0.9-SNAPSHOT.pom
	[warn] ==== local-preloaded-ivy: tried
	[warn]   C:\Users\tschoening\.sbt\preloaded\io.kaitai\kaitai-struct-compiler_2.12\0.9-SNAPSHOT\ivys\ivy.xml
	[warn] ==== local-preloaded: tried
	[warn]   file:/C:/Users/tschoening/.sbt/preloaded/io/kaitai/kaitai-struct-compiler_2.12/0.9-SNAPSHOT/kaitai-struct-compiler_2.12-0.9-SNAPSHOT.pom
	[warn] ==== Sonatype OSS Snapshots: tried
	[warn]   https://oss.sonatype.org/content/repositories/snapshots/io/kaitai/kaitai-struct-compiler_2.12/0.9-SNAPSHOT/kaitai-struct-compiler_2.12-0.9-SNAPSHOT.pom
	[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
	[warn]  ::          UNRESOLVED DEPENDENCIES         ::
	[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
	[warn]  :: io.kaitai#kaitai-struct-compiler_2.12;0.9-SNAPSHOT: not found
	[warn]  ::::::::::::::::::::::::::::::::::::::::::::::

I manually fixed this by linking my output directory for the compiler to `file:/C:/Users/tschoening/.sbt/preloaded/io/kaitai/kaitai-struct-compiler_2.12/0.9-SNAPSHOT`, but of course would avoid such manual step in the long term. Especially as it already breaks for the next release of the compiler because of version/directory name mismatch. Maybe my build setup is too customized, but from my point of view I'm simply doing everything in your relative directory structure.

**Comment by GreyCat** [comment]
> Is this KST-thing documented somewhere?

Actually, not, and this is my huge miss. Let me try to fix that by starting at least something on KST.

**Comment by ams-tschoening** [comment]
How about merging the pushed KST first? I would like to delete this branch and problems regarding missing docs etc. are anyway better managed using independent issues I guess.

**Pull request merged**